pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -201,9 +214,19 @@ func (p *staticPolicy) RemoveContainer(s state.State, containerID string) error
 	return nil
 }
 
-func (p *staticPolicy) allocateCPUs(s state.State, numCPUs int) (cpuset.CPUSet, error) {
-	klog.Infof("[cpumanager] allocateCpus: (numCPUs: %d)", numCPUs)
-	result, err := takeByTopology(p.topology, p.assignableCPUs(s), numCPUs)
+func (p *staticPolicy) allocateCPUs(s state.State, numCPUs int, sockets map[int]bool) (cpuset.CPUSet, error) {
+	klog.Infof("[cpumanager] allocateCpus: (numCPUs: %d, socket: %d)", numCPUs, sockets)
+	assignableCPUs := cpuset.NewCPUSet()
+	if len(sockets) != 0 {
+		for socketID, socketAvail := range sockets {
+			if socketAvail {
```

Comment:
When is this ever not true?

Reply:
sockets would be nil if TopologyHints are nil for the container. Should never really happen, but gives extra security.